src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -204,6 +230,51 @@ protected override void Free()
             base.Free();
         }
 
+        protected override bool TryGetVariable(VariableIdentifier identifier, out int slot)
+        {
+            return _variableSlot.TryGetValue(identifier, out slot);
+        }
+
+        protected override int AddVariable(VariableIdentifier identifier)
+        {
+            int slot = nextVariableSlot++;
+            _variableSlot.Add(identifier, slot);
+            while (slot >= variableBySlot.Count)
+            {
+                variableBySlot.Add(default);
```

Comment:
Consider using `AddMany`.

Reply:
`slot` should be at most equal to, not greater than, `variableBySlot.Count` so the `while` loop is executed at most once. And the alternative seems more complicated: `if (slot >= variableBySlot.Count) variableBySlot.AddMany(default, variableBySlotCount - slot + 1);`

---
In reply to: [560532232](https://github.com/dotnet/roslyn/pull/50417#discussion_r560532232) [](ancestors = 560532232)